style_guides/accessibility_guide.md

```diff
@@ -0,0 +1,166 @@
+
+### tabindex and id
+
+**TL;DR** *Only use `tabindex="0"` and `tabindex="-1"` and no values above 0. Always
+add an `id` to an element with `tabindex`.*
+
+If you want to make an element focusable, that doesn't offer this feature by default
+(i.e. isn't an form element or a link), you can add `tabindex` with a value >= 0 to it.
+
+*Why shouldn't you use values above 0?*
+
+Setting `tabindex="0"` will add the element to the tabbing order at the position
+it is in the DOM. Setting it to something above 0 will create an explicit order.
+Tabbing will always focus all elements with an tabindex > 0, starting from the smallest
+number. After all those elements has been tabbed, it will continue with all `tabindex="0"`
+or implicit tabable elements. These values are not scoped to a subtree of the
+DOM, but are global values. Maintaining a global order is nearly impossible
+and considered a [serious issue](https://dequeuniversity.com/rules/axe/1.1/tabindex)
+by automated accessibility testing frameworks.
+
+`tabindex="-1"` will remove an element from tab order, that would be focusable
+otherwise. You won't need this often.
+
+*Why should you add an `id`?*
+
+Due to some bugs in some common screen readers, you will always need to add an `id`
+to an element with `tabindex`, since they wouldn't pick up the `tabindex` correctly
+otherwise.
+
+### Tooltips
+
+**TL;DR** *Add `role="tooltip"` and `aria-describedby` to elements for accessible tooltips.*
+
+Elements which act as tooltips should have the `role="tooltip"` attribute and an ID to which the
+described element can point to with the `aria-describedby` attribute. For example:
+
+```html
+<div
+  class="kuiTooltip"
+  role="tooltip"
+  id="visualizationsTooltip"
+>
+  Visualizations help you make sense of your data.
+</div>
+
+<button aria-describedby="visualizationsTooltip">
+ Visualizations
+</button>
+```
+
+### Don't create keyboard traps
+
+**TL;DR** *If you can't leave an element with <kbd>Tab</kbd> again, it needs a special interaction model.*
+
+If an interactive element consumes the <kbd>Tab</kbd> key (e.g. a code editor to
+create an actual tabular indentation) it will prevent a keyboard user to leave
+that element again. Also see [WCAG 2.1.2](https://www.w3.org/TR/WCAG20/#keyboard-operation-trapping).
+
+Those kind of elements, require a special interaction model. A [code editor](https://github.com/elastic/kibana/pull/13339)
+could require an <kbd>Enter</kbd> keypress before starting editing mode, and
+could leave that mode on <kbd>Escape</kbd> again.
+
+Unfortunately there is no universal solution for this problem, so be aware when creating
+such elements, that would consume tabbing, to think about an accessible interaction
+model.
+
+*Hint:* If you create that kind of interactive elements `role="application"` might
+be a good `role` (also see below) for that element. It is meant for elements providing
+their own interaction schemes.
+
+## Roles
```

Comment:
There's a very ambiguous line between "complete guideline" and "common pitfalls" because it's so relative to the experience level of the developer. I think we might as well err on the side of comprehensiveness and describe those roles, as well as `aria-label/aria-labelledby` and `aria-describedby`. We can use https://github.com/elastic/kibana/issues/11837#issuecomment-308510404 for the latter.